### PR TITLE
fix pad to square

### PR DESCRIPTION
**Pull request by grimoire**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily receiving feedbacks. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Yolox use pad mode `pad_to_square`, which is conflict with current object detection task preprocess.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit tests to ensure the correctness.
3. If the modification has a dependency on downstream projects of a newer version, this PR should be tested with all supported versions of downstream projects.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

**Review by AllentDan — APPROVED** [review]
LGTM

File: mmdeploy/codebase/mmdet/deploy/object_detection.py [hunk 1]
```diff
@@ -122,22 +125,12 @@ def create_input(self,
         from mmdet.datasets.pipelines import Compose
         if isinstance(imgs, (str, np.ndarray)):
             imgs = [imgs]
-        dynamic_flag = is_dynamic_shape(self.deploy_cfg)
         model_cfg = self.model_cfg
         if pipeline_updater is not None:
             model_cfg = pipeline_updater(self.deploy_cfg, model_cfg)
         cfg = process_model_config(model_cfg, imgs, input_shape)
         # Drop pad_to_square when static shape. Because static shape should
```
Comment (hanrui1sensetime) [inline mmdeploy/codebase/mmdet/deploy/object_detection.py:132]: Please remove this comment respectively.

**Comment by hanrui1sensetime** [comment]
Seems ut failed, may fix ut.

**Review by hanrui1sensetime — APPROVED** [review]
LGTM.

**Pull request merged**